src/app/rpmostree-builtin-start-daemon.c

```diff
@@ -50,40 +58,43 @@ static GOptionEntry opt_entries[] =
 static RpmostreedDaemon *rpm_ostree_daemon = NULL;
 
 static void
-start_daemon (GDBusConnection *connection)
+state_transition (AppState state)
```

Comment:
Could we make this take a string as well to print to the journal as to why we're transitioning to a new state? E.g.

> Transitioning state (RUNNING --> FLUSHING): Caught signal

I think NetworkManager does something similar as well (though there, adapters/connections can go back and forth between states, rather than being linear -- but it's a nice way to describe what's going on).